app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1682,42 +1696,79 @@ public void showControlsThenHide() {
-        showOrHideButtons();
-        showSystemUIPartially();
-
         final int hideTime = binding.playbackControlRoot.isInTouchMode()
-                ? DEFAULT_CONTROLS_HIDE_TIME
-                : DPAD_CONTROLS_HIDE_TIME;
-
-        showHideShadow(true, DEFAULT_CONTROLS_DURATION);
-        animate(binding.playbackControlRoot, true, DEFAULT_CONTROLS_DURATION,
-                AnimationType.ALPHA, 0, () -> hideControls(DEFAULT_CONTROLS_DURATION, hideTime));
+            ? DEFAULT_CONTROLS_HIDE_TIME
+            : DPAD_CONTROLS_HIDE_TIME;
+        showControls(
+            DEFAULT_CONTROLS_DURATION,
+            () -> hideControls(DEFAULT_CONTROLS_DURATION, hideTime)
+        );
     }
 
     public void showControls(final long duration) {
+        showControls(duration, null);
+    }
+
+    public void showControls(final long duration, final Runnable callback) {
         if (DEBUG) {
             Log.d(TAG, "showControls() called");
         }
-        showOrHideButtons();
-        showSystemUIPartially();
+
+        final RelativeLayout layoutToShow;
+        if (!isLocked) {
+            showOrHideButtons();
+            showSystemUIPartially();
+            layoutToShow = binding.playbackControlRoot;
+        } else {
+            layoutToShow = binding.lockedRoot;
+        }
+
         controlsVisibilityHandler.removeCallbacksAndMessages(null);
         showHideShadow(true, duration);
-        animate(binding.playbackControlRoot, true, duration);
+        animate(
+            layoutToShow,
+            true,
+            duration,
+            AnimationType.ALPHA,
+            0,
+            callback
+        );
     }
 
     public void hideControls(final long duration, final long delay) {
+        hideControls(duration, delay, null);
+    }
+
+    public void hideControls(final long duration, final long delay, final Runnable callback) {
         if (DEBUG) {
             Log.d(TAG, "hideControls() called with: duration = [" + duration
                     + "], delay = [" + delay + "]");
         }
 
-        showOrHideButtons();
+        final RelativeLayout layoutToHide;
+        if (!isLocked) {
+            showOrHideButtons();
+            layoutToHide = binding.playbackControlRoot;
+        } else {
+            layoutToHide = binding.lockedRoot;
+        }
 
         controlsVisibilityHandler.removeCallbacksAndMessages(null);
         controlsVisibilityHandler.postDelayed(() -> {
             showHideShadow(false, duration);
-            animate(binding.playbackControlRoot, false, duration, AnimationType.ALPHA,
-                    0, this::hideSystemUIIfNeeded);
+            animate(
+                layoutToHide,
+                false,
+                duration,
+                AnimationType.ALPHA,
+                0,
+                () -> {
+                    hideSystemUIIfNeeded();
+                    if (callback != null) {
+                        callback.run();
+                    }
+                }
```

Comment:
Same here with the callback

Reply:
This is used when pressing the lock button. The UI is hidden, and then the locked state is set to true right after.